### PR TITLE
Return the error cases in runTxSubmit

### LINKED ISSUE #1118: [BUG] - cardano-cli does not exit with correct errorcode
Tag: pioneer-wave2

Tried to submit a stakepool de-registration certificate on the chain, cli exited with the following message:
`ApplyTxError [LedgerFailure (DelegsFailure (DelplFailure (PoolFailure (StakePoolRetirementWrongEpochPOOL {pfPOOLcurrentEpoch = 32, pfPOOLintendedEpoch = 35, pfPOOLepochMax = 33}))))]`

**But the exitcode (echo $?) of cardano-cli was still 0 (no Error) !**

I haven't checked other cardano-cli errors, but cardano-cli should(must) always exit with an exitcode other then 0 if something went wrong. Otherwise it's a nightmare to integrate it safe in automated scripts.

Thx!

### PR DESCRIPTION
**Pull request by intricate**
Closes #1118

### CODE REVIEW

File: cardano-cli/src/Cardano/CLI/Shelley/Run/Transaction.hs [hunk 1]
```diff
@@ -120,10 +123,9 @@ runTxSubmit txFp network = do
   signedTx <- firstExceptT ShelleyTxReadSignedTxError . newExceptT $ readTxSigned txFp
   result   <- liftIO $ submitTx network sktFp signedTx
   case result of
-    TxSubmitSuccess            -> return ()
-    --TODO: use the Cardano.Api.TxSubmit.ErrorRender here
-    TxSubmitFailureByron   err -> liftIO $ print err
-    TxSubmitFailureShelley err -> liftIO $ print err
+    TxSubmitSuccess          -> return ()
+    TxSubmitFailureShelley _ -> left (ShelleyTxSubmitError result)
+    TxSubmitFailureByron   _ -> left (ShelleyTxSubmitError result)
```
Comment (intricate) [inline cardano-cli/src/Cardano/CLI/Shelley/Run/Transaction.hs:128]: Is it not odd that we're able to submit Byron transactions from this command?

File: cardano-cli/src/Cardano/CLI/Shelley/Run/Transaction.hs [hunk 2]
```diff
@@ -69,6 +70,8 @@ renderShelleyTxCmdError err =
       "Error while writing signed shelley tx: " <> renderApiError apiError
     ShelleyTxWriteUnsignedTxError apiError ->
       "Error while writing unsigned shelley tx: " <> renderApiError apiError
+    ShelleyTxSubmitError txSubmitRes ->
+      "Error while submitting Shelley tx: " <> textShow txSubmitRes
```
Comment (Jimbo4350) [inline cardano-cli/src/Cardano/CLI/Shelley/Run/Transaction.hs:74]: We should write a `renderTxSubmitError` function here to provide more context with the errors vs just showing them.

File: cardano-cli/src/Cardano/CLI/Shelley/Run/Transaction.hs (hunk 1 above)
Comment (Jimbo4350) [reply cardano-cli/src/Cardano/CLI/Shelley/Run/Transaction.hs:128]: This should get sorted out as we flesh out the typed API

**Comment by dcoutts** [comment]
bors merge

**Comment by dcoutts** [comment]
bors merge

**Pull request merged**